packages/client/hmi-client/src/components/workflow/tera-simulate-node.vue

```diff
@@ -0,0 +1,242 @@
+				/>
+			</div>
+		</div>
+		<div class="result">
+			<Chart type="line" :data="chartData" :options="chartOptions" />
+		</div>
+	</section>
+</template>
+
+<script setup lang="ts">
+import { ref, watch } from 'vue';
+import { PetriNet } from '@/petrinet/petrinet-service';
+import { ITypedModel } from '@/types/Model';
+import Button from 'primevue/button';
+import { csvParse } from 'd3';
+
+import MultiSelect from 'primevue/multiselect';
+import Chart from 'primevue/chart';
+
+import { makeForecast, getRunStatus, getRunResult } from '@/services/models/simulation-service';
+import { WorkflowNode } from '@/types/workflow';
+
+const props = defineProps<{
+	node: WorkflowNode;
+}>();
+
+const startedRunIdList = ref([] as number[]);
+const completedRunIdList = ref([] as number[]);
+let runResults = {};
+
+// data for rendering ui
+let stateVariablesList = [] as any[];
+const selectedVariable = ref<{ code: string }[]>([]);
+let runList = [] as any[];
+const selectedRun = ref<null | { code: string }>(null);
+
+const chartData = ref({});
+const chartOptions = {
+	maintainAspectRatio: false,
+	pointStyle: false,
+	plugins: {
+		legend: {
+			labels: {
+				color: '#000'
+			}
+		}
+	},
+	scales: {
+		x: {
+			ticks: {
+				color: '#000'
+			},
+			grid: {
+				color: '#AAA'
+			}
+		},
+		y: {
+			ticks: {
+				color: '#000'
+			},
+			grid: {
+				color: '#AAA'
+			}
+		}
+	}
+};
+
+// FIXME: adapt to new model representation
+// FIXME: adapt to new simulation-service id-based API
+const scrubModel = (model: ITypedModel<PetriNet>) => {
+	const cleanedModel: PetriNet = {
+		S: [],
+		T: [],
+		I: [],
+		O: []
+	};
+	if (model) {
+		cleanedModel.S = model.content.S.map((s) => ({ sname: s.sname }));
+		cleanedModel.T = model.content.T.map((t) => ({ tname: t.tname }));
+		cleanedModel.I = model.content.I;
+		cleanedModel.O = model.content.O;
+	}
+	return JSON.stringify(cleanedModel);
+};
+
+const runSimulate = async () => {
+	const port = props.node.inputs[0];
+	if (port && port.value) {
+		const payload = {
+			model: scrubModel(port.value.model),
+			initials: port.value.initialValues,
+			params: port.value.parameterValues,
+			tspan: [0, 100]
+		};
+
+		const response = await makeForecast(payload);
+		startedRunIdList.value = [response.id];
+
+		// start polling for run status
+		getStatus();
+	}
+};
+
+// watch for changes in node input
+// watch(
+// 	() => props.node.inputs,
+// 	async (inputList) => {
+// 		const forecastOutputList = await Promise.all(
+// 			inputList.map(({ value }) =>
+// 				makeForecast({
+// 					model: value.model.id,
+// 					initials: value.initialValues,
+// 					params: value.parameterValues,
+// 					tspan: [0.0, 90.0] // hardcoded timespan
+// 				})
+// 			)
+// 		);
+// 		startedRunIdList.value = forecastOutputList.map((forecastOutput) => forecastOutput.id);
+//
+// 		// start polling for run status
+// 		getStatus();
+// 	},
+// 	{ deep: true }
+// );
+
+// Retrieve run ids
+// FIXME: Replace with API.poller
```

Comment:
Then please do.

Reply:
this can be fixed later, will need to retest after replace with API.poller